File: backend/domain/entities/repair_order.py
from dataclasses import dataclass
from datetime import date
from typing import List
from domain.entities.order_detail import OrderDetail

@dataclass
class RepairOrder:
    id_vehiculo: int
    fecha_inicio: date
    estado: str
    mano_de_obra: float
    prioridad: int
    details: List[OrderDetail]
    id: int | None = None
    fecha_fin: date | None = None

File: frontend/src/domain/entities/repairOrder.js
export class RepairOrder {
    constructor({ id, id_vehiculo, fecha_inicio, fecha_fin, estado, mano_de_obra, prioridad, details }) {
      this.id = id;
      this.id_vehiculo = id_vehiculo;
      this.fecha_inicio = fecha_inicio;
      this.fecha_fin = fecha_fin;
      this.estado = estado;
      this.mano_de_obra = mano_de_obra;
      this.prioridad = prioridad;
      this.details = details || [];
    }
  }